### ISSUE TITLE
has_many association throwing 'stack level too deep'

### ISSUE THREAD

**Issue by chadwtaylor** [remiprev/her#385]
I've google and also tried to isolate the issue on my own before resorting to create an issue here.

When I add `has_many :services` to the Person model, and calling this `@person.services` would result in `stack level too deep`. Every time I remove `has_many :services`, it works.

It is worth mentioning that I use UUIDs as ids (eg: `service_id: "5ab5e60a-331e-63cf-b2aa-757e8cc316dc"`).

Is anybody aware of this issue -- and I'm wondering if it has to do with a string-based UUID vs an integer-based ID?

```ruby
class Person
  include Her::Model

  has_many :services
end
```

```ruby
class Service
  include Her::Model

  belongs_to :person # also tried has_one but same issue occured
end
```

Thanks in advance.

**Comment by alex-bezek** [comment]
What was the resolution for this? I am having the same issue

**Comment by alex-bezek** [comment]
ok, I ended up resolving this for my situation. I have a kind of theory as to why it was behaving this way, but i'm not 100% sure. 
my route: /organizations/:organization_id/user
its important to note that while experiencing this error, my api did not return the organization_id (adding this to the api response fixed it)

User class has_many settings.
I noticed when getting my results from her, if I specified one of the has_many objects after I got my result, it would also have this backwards link to the whole object
`result = HerClient::User.where(organization_id: 'test').all`
`result.first.settings.first`
 would not only show the data for the setting, but also include a link to the whole user object. I think when the response didn't include the organization_id, it would try to get this user parent object, not have the organization id, and freak out. Adding organization_id allowed it to figure out that the parent object User had a list of settings, but these are the same settings it just queried for and to stop the recursive process.

If anyone has another idea, I would be curious. If someone wants, I can make a gist or small project they could test against. 

Thanks
